### PR TITLE
UCT/RC: Check AM ordering

### PR DESCRIPTION
**Pull request by brminich**
## What
Add ordering checks for UCT AM APIs in RC transports. Check is not added for other communication functions, because ordering between them and AM funcs is not respected (others are not affected by  FC).

## Why ?
To avoid future changes which would ruin ordering between AM messages.

### CODE REVIEW

**Comment by brminich** [comment]
issue is #5765

File: src/uct/ib/rc/base/rc_iface.h [hunk 1]
```diff
@@ -493,4 +496,34 @@ uct_rc_iface_fence_relaxed_order(uct_iface_h tl_iface)
 
     return uct_rc_iface_fence(tl_iface, 0);
 }
+
+#if UCS_ENABLE_ASSERT
+static UCS_F_ALWAYS_INLINE int
+uct_rc_iface_check_pending(uct_rc_iface_t *iface, ucs_arbiter_group_t *arb_group)
+{
+    return (iface->tx.in_pending || ucs_arbiter_group_is_empty(arb_group));
```
Comment (yosefe) [inline src/uct/ib/rc/base/rc_iface.h:504]: maybe do ucs_assert inside this function? would remove some code, and can remove `#if` from line 500

File: src/uct/ib/rc/accel/rc_mlx5_ep.c [hunk 2]
```diff
@@ -370,6 +367,8 @@ ucs_status_t uct_rc_mlx5_ep_am_zcopy(uct_ep_h tl_ep, uint8_t id, const void *hea
         UCT_TL_EP_STAT_OP(&ep->super.super, AM, ZCOPY,
                           header_length + uct_iov_total_length(iov, iovcnt));
         UCT_RC_UPDATE_FC(&iface->super, &ep->super, id);
+        ucs_assert(uct_rc_iface_check_pending(&iface->super,
+                                              &ep->super.arb_group));
```
Comment (yosefe) [inline src/uct/ib/rc/accel/rc_mlx5_ep.c:371]: maybe move UCT_RC_CHECK_RES outside of uct_rc_mlx5_ep_zcopy_post, and in line 359 replace by UCT_RC_CHECK_RES_AND_FC ?

**Review by yosefe — APPROVED** [review]

**Pull request merged**